### PR TITLE
Fix real-time telemetry

### LINKED ISSUE #144: Realtime telemetry completely broken in 2.0.5
#### Summary
No subscriptions are being made for any parameters. This is ~~almost~~ certainly due to [recent refactoring of WebSockets](https://github.com/akhenry/openmct-yamcs/pull/127/files#diff-4da409899847f2633a97c8906bc2948b6a2534c8ce6d3a635224ddb063c36ac4R1)

#### Expected vs Current Behavior
Clicking on a parameter should show data arriving in real-time.

#### Impact Check List

- [x] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?

#### Steps to Reproduce

1. Click on any parameter with real-time data flowing.
2. Observe that data is not displayed or updated in real-time.

### PR DESCRIPTION
**Pull request by akhenry**
Closes #144

### Describe your changes:
* Restores the previous realtime-telemetry-provider which is battle-hardened and known to handle WebSocket dropouts reliably. 
* Refactors the realtime-fault-provider to make use of the realtime-telemetryprovider.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/providers/realtime-provider.js [hunk 1]
```diff
@@ -86,98 +99,172 @@ export default class RealtimeProvider {
...
+        if (this.reconnectTimeout) {
             return;
         }
 
-        let isReply = data.type === DATA_TYPES.DATA_TYPE_REPLY;
-        let subscriptionDetails;
+        this.reconnectTimeout = setTimeout(() => {
+            this.connect();
+            delete this.reconnectTimeout;
+        }, FALLBACK_AND_WAIT_MS[this.currentWaitIndex]);
```
Comment (jvigliotta) [inline src/providers/realtime-provider.js:125]: This is cool!

File: src/providers/realtime-provider.js [hunk 2]
```diff
@@ -86,98 +99,172 @@ export default class RealtimeProvider {
...
+            try {
+                this.sendMessage(request);
+                return true;
+            } catch (error) {
+                this.connected = false;
+                console.error(error);
+                console.warn("Error while attempting to send to websocket. Reconnecting...");
+
+                this.requests.push(request);
+                this.reconnect();
```
Comment (jvigliotta) [inline src/providers/realtime-provider.js:143]: Do we need to return false here? I see you're returning true above. Is anything relying on this? I didn't see anything, if that's true, you may be able to remove that `return true` above.

File: src/providers/realtime-provider.js [hunk 3]
```diff
@@ -86,98 +99,172 @@ export default class RealtimeProvider {
...
+    }
+
+    connect() {
+        if (this.connected) {
+            return;
+        }
+        let wsUrl = `${this.url}`;
+        this.lastSubscriptionId = 1;
+        this.connected = false;
+        this.socket = new WebSocket(wsUrl);
```
Comment (jvigliotta) [inline src/providers/realtime-provider.js:157]: I know we're overwriting this here, but do we need to set to undefined or anything first? I know we have listeners attached, I think they get removed on error or close, but just checking.

File: src/providers/realtime-provider.js [hunk 4]
```diff
@@ -86,98 +99,172 @@ export default class RealtimeProvider {
...
         });
     }
 
-    _sendSubscribeMessage(subscriptionDetails) {
-        let domainObject = subscriptionDetails.domainObject;
-        let message = MESSAGES.SUBSCRIBE[domainObject.type](subscriptionDetails);
+    flushQueue() {
+        let shouldReconnect = false;
+        this.requests = this.requests.filter((request) => {
+            try {
```
Comment (jvigliotta) [inline src/providers/realtime-provider.js:244]: Cool use of filter here.

**Review by jvigliotta — CHANGES_REQUESTED** [review]
Lookin nice! Lot's of cool stuff in this. Seems pretty rock solid. Just had a couple suggestions and questions, nothing major.

**Comment by akhenry** [comment]
@jvigliotta I believe I have addressed all of these comments, but due to what appears to be a GitHub bug all of my comments are stuck in a `pending` state.

**Pull request merged**